### PR TITLE
feat(Anchor): Add new anchor component

### PR DESCRIPTION
**Pull request by mayank99**
Added missing anchor component using global `iui-anchor` class.

image

## Checklist

- [x] Add meaningful unit tests for your component (verify that all lines are covered)
- [x] Verify that all existing tests pass
- [x] Add component features demo in Storybook (different stories)
- [x] Approve test images for new stories
- [x] Add screenshots of the key elements of the component

### CODE REVIEW

File: stories/core/Typography/Anchor.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  component: Anchor,
+  parameters: {
+    controls: { hideNoControlsWarning: true },
+  },
+  argTypes: {},
+} as Meta;
+
+export const Basic: Story<React.ComponentPropsWithoutRef<'a'>> = (args) => {
+  return (
+    <Anchor href='https://www.example.com/' {...args}>
```
Comment (bentleyvk) [inline stories/core/Typography/Anchor.stories.tsx:20]: Maybe better change to `www.bentley.com`.

File: stories/core/Typography/Anchor.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      www.example.com
```
Comment (bentleyvk) [inline stories/core/Typography/Anchor.stories.tsx:21]: Maybe change to `Bentley.com`

File: stories/core/Typography/Anchor.stories.tsx (hunk 1 above)
Comment (mayank99) [reply stories/core/Typography/Anchor.stories.tsx:20]: How is that better?

Comment (bentleyvk) [reply stories/core/Typography/Anchor.stories.tsx:20]: Not directing to random URL which might take you anywhere.

Comment (mayank99) [reply stories/core/Typography/Anchor.stories.tsx:20]: But it's not a random url. [example.com](https://www.example.com) is an officially reserved URL for documentation/demo purposes

**Review by saaaaaally — APPROVED** [review]

**Review by bentleyvk — APPROVED** [review]

**Pull request merged**